legend-engine-xt-serviceStore-pure/src/main/resources/core_servicestore/metamodel/serviceStore.pure

```diff
@@ -122,8 +123,46 @@ Class <<typemodifiers.abstract>> meta::external::store::service::metamodel::Secu
 {
 }
 
+Class meta::external::store::service::metamodel::SimpleHttpSecurityScheme extends meta::external::store::service::metamodel::SecurityScheme
+{
+  scheme : String[1];
+}
+
+Class meta::external::store::service::metamodel::ApiKeySecurityScheme extends meta::external::store::service::metamodel::SecurityScheme
+{
+  location : ApiKeyLocation[1];
+  keyName  : String[1];
+}
+
+Enum meta::external::store::service::metamodel::ApiKeyLocation
+{
+   COOKIE,
+   HEADER
+}
+
 Class meta::external::store::service::metamodel::SerializationFormat
 {
    style   : String[0..1];
    explode : Boolean[0..1];
+}
+
+Class <<typemodifiers.abstract>> meta::external::store::service::metamodel::SecuritySchemeRequirement
+{
+}
+
+Class meta::external::store::service::metamodel::SingleSecuritySchemeRequirement extends meta::external::store::service::metamodel::SecuritySchemeRequirement
+{
+    id               : String[1];
+    securityScheme   : SecurityScheme[1];
+}
+
+Class meta::external::store::service::metamodel::MultiSecuritySchemeRequirement extends meta::external::store::service::metamodel::SecuritySchemeRequirement
+{
+    operation        : Operation[1];
+    securitySchemes  : SingleSecuritySchemeRequirement[*];
+}
+
+Enum meta::external::store::service::metamodel::Operation
```

Comment:
Can we ever have anything other than multiSecuritySchemeRequirement?
If not maybe we can remove it from here and we will have it in CompositeAuth which would not be limited to service store

Reply:
We need some wrapper to hold which security schemes to use together in case of multiple needs to be used. CompositeAuth will only hold corresponding authentication specirfications.

Reply:
Apologies I meant can we have any other operation other than AND. I guess no so lets remove it